scripts/tests/test_zkai_minimal_d128_block_boundary_wrapper_gate.py

```
@@ -0,0 +1,274 @@
+        self.assertEqual(rows["minimal_wrapper_boundary"]["saving_bytes"], 0)
+        self.assertEqual(
+            rows["minimal_wrapper_boundary"]["typed_bytes"],
+            len(gate.canonical_json_bytes(self.payload["boundary_statement"])),
+        )
+        self.assertEqual(rows["minimal_wrapper_boundary"]["commitment"], self.payload["boundary_statement_commitment"])
+
+    def test_malformed_source_rows_reject_with_gate_error(self) -> None:
+        sources, _descriptors = gate.load_sources()
+        missing_summary = copy.deepcopy(sources)
+        missing_summary["model_faithful_single"].pop("summary")
+        with self.assertRaisesRegex(gate.MinimalD128BlockBoundaryWrapperError, "model faithful single summary"):
```

Ruff (0.15.13)

[warning] Use `pytest.raises` instead of unittest-style `assertRaisesRegex`

Replace `assertRaisesRegex` with `pytest.raises`

(PT027)